src/box/tuple_compare.cc

```diff
@@ -1400,13 +1400,17 @@ static const comparator_with_key_signature cmp_wk_arr[] = {
 
 /**
  * A functional index tuple compare.
- * tuple_a_hint and tuple_b_hint are expected to be valid
- * pointers to functional key memory. These keys have been already
- * validated and are represented as a MsgPack array with exactly
- * func_index_part_count parts. The cmp_def has part_count > func_index_part_count,
- * since it was  produced by key_def_merge() of the functional key part
- * and the primary key. So its tail parts are taken from primary
- * index key definition.
+ * tuple_a_hint and tuple_b_hint are expected to be valid pointers to functional
+ * key memory. These keys are represented as a MsgPack array with the number of
+ * elements equals to the functional index definition part_count (let's denote
+ * it as func_index_part_count). The keys have been already validated by
```

Comment:
Nit: it's pretty confusing, that we expect arrays with the same number of elements, but later we take a minimum (`uint32_t key_part_count = MIN(part_count_a, part_count_b);`). This was written even before this patch, but while we're at it, it would be nice to do an equality check in an assertion, and take one of the part counts.

Reply:
Done.